src/python_bindings/py_algorithm.h

```diff
@@ -3,13 +3,53 @@
 #include <memory>
 #include <type_traits>
 #include <unordered_map>
+#include <vector>
 
 #include <pybind11/pybind11.h>
 #include <pybind11/stl.h>
 
 #include "algorithms/algorithm.h"
+#include "algorithms/metric/enums.h"
+#include "config/equal_nulls/type.h"
+#include "config/error/type.h"
+#include "config/indices/type.h"
+#include "config/max_lhs/type.h"
 #include "config/tabular_data/input_table_type.h"
+#include "config/thread_number/type.h"
+#include "model/types/types.h"
+namespace {
+using ConvFunction = std::function<std::string(boost::any)>;
+template <typename T>
+std::pair<std::string, ConvFunction> generate_conv_pair(std::string name) {
+    return std::pair<std::string, ConvFunction>{
+            name, [](boost::any value) { return std::to_string(boost::any_cast<T>(value)); }};
+}
+
+template <>
+std::pair<std::string, ConvFunction> generate_conv_pair<bool>(std::string name) {
+    return std::pair<std::string, ConvFunction>{
+            name, [](boost::any value) { return boost::any_cast<bool>(value) ? "True" : "False"; }};
+}
+const std::unordered_map<std::string, ConvFunction> converters{
```

Comment:
Put these conversions into a separate file, like `PyToAny`.

Reply:
And I'm pretty sure you can return Python objects with a proper type that aren't only `str` if you simply convert `any` to its actual type then wrap it into `py::object` in these converter functions, relying on pybind11's conversions